### ISSUE TITLE
Bug with aws_spot_fleet_request and IAM Instance Profiles

### ISSUE THREAD

**Issue by justintime** [hashicorp/terraform#9875]
### Terraform Version
0.7.8

### Affected Resource(s)
- aws_spot_fleet_request

### Problem Description
Although the documentation is fairly vague on what all is supported in the launch_specification, the AWS docs that are linked seem to imply that you can use ```iam_instance_profile``` to add profiles to launched instances.  Passing in the name of the profile seems to result in the proper actions:
```
  launch_specification.3796383985.iam_instance_profile:                      "" => "arn:aws:iam::23983246924:instance-profile/tf-ephemeral-ec2-instance"
```
However, the resulting spot fleet request doesn't have any IAM Profile attached to it in the console, nor do any of the resulting ec2 instances have the profile assigned to it.

Just on a hunch, I tried passing in the ARN of the profile, and that did the exact same thing -- appeared to work, but didn't actually do anything.

### References
This was mentioned on the initial pull request by another user here: https://github.com/hashicorp/terraform/issues/4070#issuecomment-189920173

Pinging @solarkennedy to see if he's come across this problem or not.

**Comment by solarkennedy** [comment]
I'm pretty sure this works, we use this feature at Yelp.
On spot fleet though, the actual EC2 launch api requests come *from* the spot fleet service, so sometimes the actual errors show up in the ec2 console, and can't really be revealed by TF.

AFAIK the actual string to pass is `tf-ephemeral-ec2-instance`, not the full ARN.

Can you look at the EC2 console and see if there is a smoking gun in the history thing?

**Comment by justintime** [comment]
Hmm, I'm not having any luck.

TF *reports* that it's assigning the profile to the fleet request, but the console doesn't show it.

I stripped away a lot of extra properties not needed for testing and ran this code:
```hcl
resource "aws_spot_fleet_request" "site" {
  iam_fleet_role = "${aws_iam_role.spot-fleet-role.arn}"
  spot_price = "1.05"
  target_capacity = 2
  valid_until = "2036-11-04T20:44:20Z"
  terminate_instances_with_expiration = "true"

  launch_specification {
    instance_type           = "m3.xlarge"
    ami                     = "${var.site_ami_id}"
    iam_instance_profile    = "tf-ephemeral-ec2-instance"
    key_name                = "${aws_key_pair.chef_provisioning.key_name}"
    subnet_id               = "${module.vpc.app_subnets[0]}"
  }
}
```

Running that resulted with this in the console:

![screenshot 2016-11-07 10 44 52](https://cloud.githubusercontent.com/assets/144090/20066616/bc96b8b2-a4d7-11e6-8d55-e21cbcb93f3a.png)

Since the profile isn't assigned to the fleet request, the individual spot requests "succeed", but lack the profile:

![screenshot 2016-11-07 10 49 18](https://cloud.githubusercontent.com/assets/144090/20066661/f6fe59b0-a4d7-11e6-8682-dc9fb6461c60.png)

Do you see anything I might be doing wrong or significantly different?

**Comment by solarkennedy** [comment]
Can you humor me and look at the dashboard at an *actual* instance and see if it has a `tf-ephemeral-ec2-instance` IAM role attached to it?

Mine do, even though on the SFR page it just has "IAM role: -` like yours. :(

**Comment by justintime** [comment]
Oh man...

You're right, the instance does have the proper profile assigned.

I think I may have been chasing a red-herring this whole time.  I saw an issue that could have been caused by a missing profile, and saw the "-" in the console, and figured that had to be it.  When I created a fleet request via the console and saw the profile listed on the SFR, I figured I'd found my problem.

It might be a QOL improvement to figure out how to make it show up on the SFR, but obviously, that's a nice-to-have.

Sorry about all this!!
